code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -143,10 +142,6 @@ private void activateServices(final Service initialService, final List<Service>
          */
         try {
             for (Service service : services) {
-                if (service.getId().equals(initialService.getId())) {
```

Comment:
I haven't looked at the rest of the changes yet, but was curious about this change.